source/shared/cpp/ObjectModel/ParseUtil.h

```diff
@@ -184,4 +187,74 @@ namespace AdaptiveSharedNamespace
         T result = jsonObject.empty() ? defaultValue : deserializer(jsonObject, defaultValue);
         return result;
     }
+
+    template<typename T>
+    std::vector<std::shared_ptr<T>> ParseUtil::GetElementCollection(bool isTopToBottomContainer,
+                                                                    ParseContext& context,
+                                                                    const Json::Value& json,
+                                                                    AdaptiveCardSchemaKey key,
+                                                                    bool isRequired,
+                                                                    const std::string& impliedType)
+    {
+        auto elementArray = GetArray(json, key, isRequired);
+
+        std::vector<std::shared_ptr<T>> elements;
+        if (elementArray.empty())
+        {
+            return std::move(elements);
+        }
+
+        const size_t elemSize = elementArray.size();
+        elements.reserve(elemSize);
+
+        const ContainerBleedDirection previousBleedState = context.GetBleedDirection();
+
+        size_t currentIndex = 0;
+        for (auto& curJsonValue : elementArray)
+        {
+            ContainerBleedDirection currentBleedState = previousBleedState;
+
+            if (currentIndex != 0)
+            {
+                // In a top to bottom container (Container, Column) only the first item can bleed up, in a left
+                // to right container (ColumnSet) only the first item can bleed left
+                currentBleedState &= isTopToBottomContainer ? ~ContainerBleedDirection::BleedUp : ~ContainerBleedDirection::BleedLeft;
```

Comment:
This line was a bit confusing because just looking at the enum name and the name of the type, it didn't readily connect that it's turning off one bit.

Reply:
I'll update the comment to say that explicitly

---
In reply to: [280529780](https://github.com/Microsoft/AdaptiveCards/pull/2802#discussion_r280529780) [](ancestors = 280529780)